src/api/dependency-discovery/src/github.js

```diff
@@ -0,0 +1,66 @@
+const { request } = require('@octokit/request');
+
+// Check whether a specific time has reached the GitHub time limit.
+// @param {Date} datetime - a Date object representing a moment in the past
+// @returns {Boolean}
+const hasGitHubExpired = (dateTime) => new Date() - dateTime > 60 * 60 * 1000;
+
+const gitHubInformationRequestor = {};
+
+async function constructGitHubRequest(gitHubUrl) {
+  const labels = ['hacktoberfest', 'good first issue', 'help wanted'];
+
+  const [owner, repo] = new URL(gitHubUrl).pathname.substring(1).split('/');
```

Comment:
`new URL` can throw.

Reply:
How do we want to handle this? The exception will bubble up and then will be caught in the `try/catch` found in `router.js`. Is that what we want?

Reply:
That's fine, just potentially risky (it depends on callers doing the right thing).